### PR TITLE
remove configure script

### PR DESCRIPTION
**Pull request by siosw**
# Goals of PR

This removes the configure script and instead sets binary paths in an dotenv.
Currently only the local deploy works since the docker build is currently broken.

### CODE REVIEW

**Review by chalkandpaste — COMMENTED** [review]
This is a nice clean-up! Thanks for taking this on, @siosw! Let me know when you're ready for a review.

**Review by chalkandpaste — APPROVED** [review]

File: sbin/start_sidecar.sh [hunk 1]
```diff
@@ -1,20 +1,27 @@
...
+# currently the local sbin paths are relative to the project root
+SBIN=$(dirname "$(readlink -f "$0")")
 SBIN="`cd "$SBIN"; pwd`"
+ROOT_DIR=$SBIN/..
 
-# Check that the sidecar dotenv exists.
-ENV=".sidecar.env"
-if ! test -f $ENV; then
-    echo "Expected dotenv at $ENV (does not exist)."
+# Check that the all required dotenv files exists.
```
Comment (ujvl) [inline sbin/start_sidecar.sh:9]:
```suggestion
# Check that all required dotenv files exist.
```

File: config/local_devnet/.configure.env
Comment (ujvl) [inline config/local_devnet/.configure.env:1]: rename to something more descriptive, e.g. `.paths.env`

File: sbin/start_l1.sh [hunk 2]
```diff
@@ -1,50 +1,59 @@
...
+. $CONFIGURE_ENV
 
-# Check that the dotenv exists.
 GENESIS_ENV=".genesis.env"
 if ! test -f $GENESIS_ENV; then
     echo "Expected dotenv at $GENESIS_ENV (does not exist)."
     exit
 fi
-echo "Using genesis dotenv: $GENESIS_ENV"
+echo "Using dotenv: $GENESIS_ENV"
```
Comment (ujvl) [inline sbin/start_l1.sh:19]: nit: why change this

File: sbin/start_l1.sh [hunk 3]
```diff
@@ -1,50 +1,59 @@
...
 . $GENESIS_ENV
+
+if [ "$L1_STACK" = "geth" ]; then
 CONTRACTS_ENV=".contracts.env"
 if ! test -f $CONTRACTS_ENV; then
     echo "Expected dotenv at $CONTRACTS_ENV (does not exist)."
     exit
 fi
-echo "Using contracts dotenv: $CONTRACTS_ENV"
+echo "Using dotenv: $CONTRACTS_ENV"
```
Comment (ujvl) [inline sbin/start_l1.sh:28]: nit: why change this

File: config/local_docker/.configure.env [hunk 4]
```diff
@@ -0,0 +1,9 @@
+CONTRACTS_DIR=/specular/contracts
+CONFIG_DIR=/specular/config
+
+# Define binaries
+L1_GETH_BIN=/usr/local/bin/geth
+SP_GETH_BIN=/usr/local/bin/geth
+SP_MAGI_BIN=/usr/local/bin/magi
+SIDECAR_BIN=/usr/local/bin/sidecar
+
```
Comment (ujvl) [inline config/local_docker/.configure.env:9]: nit: extraneous extra line

File: sbin/start_l1.sh [hunk 5]
```diff
@@ -1,50 +1,59 @@
...
-SBIN=`dirname $0`
-SBIN="`cd "$SBIN"; pwd`"
-if [ -z $CONTRACTS_DIR ] | [ -z $L1_GETH_BIN ]; then
-    . $SBIN/configure.sh
+SBIN=$(dirname "$(readlink -f "$0")")
+ROOT_DIR=$SBIN/..
+
+# Check that the all required dotenv files exists.
+CONFIGURE_ENV=".configure.env"
+if ! test -f $CONFIGURE_ENV; then
```
Comment (ujvl) [inline sbin/start_l1.sh:7]:
```suggestion
if ! test -f "$CONFIGURE_ENV"; then
```

File: sbin/start_sp_geth.sh [hunk 6]
```diff
@@ -1,6 +1,28 @@
...
+CONFIGURE_ENV=".configure.env"
+if ! test -f $CONFIGURE_ENV; then
+    echo "Expected dotenv at $CONFIGURE_ENV (does not exist)."
+    exit
+fi
+echo "Using configure dotenv: $CONFIGURE_ENV"
+. $CONFIGURE_ENV
+
+SP_GETH_ENV=".sp_geth.env"
+if ! test -f $SP_GETH_ENV; then
```
Comment (ujvl) [inline sbin/start_sp_geth.sh:19]:
```suggestion
if ! test -f "$SP_GETH_ENV"; then
```

File: sbin/start_sp_magi.sh [hunk 7]
```diff
@@ -1,20 +1,27 @@
...
-if [ -z $SP_MAGI_BIN ]; then
-    # If no binary specified, assume repo directory structure.
-    . $SBIN/configure.sh
+SP_MAGI_ENV=".sp_magi.env"
+if ! test -f $SP_MAGI_ENV; then
+    echo "Expected dotenv at $SP_MAGI_ENV (does not exist)."
+    exit
 fi
-echo "Using bin: $SP_MAGI_BIN"
+echo "Using configure dotenv: $SP_MAGI_ENV"
```
Comment (ujvl) [inline sbin/start_sp_magi.sh:23]:
```suggestion
echo "Using sp-magi dotenv: $SP_MAGI_ENV"
```

**Review by ujvl — COMMENTED** [review]
can you also update the `docs/development.md` README to mention the new dotenv where the others are mentioned

File: sbin/start_sp_magi.sh [hunk 8]
```diff
@@ -1,20 +1,26 @@
...
-# Enforce that the dotenv exists.
-ENV=".sp_magi.env"
-if ! test -f $ENV; then
-    echo "Expected dotenv at $ENV (does not exist)."
+# the local sbin paths are relative to the project root
+SBIN=$(dirname "$(readlink -f "$0")")
+SBIN="`cd "$SBIN"; pwd`"
+ROOT_DIR=$SBIN/..
+
+# Check that the all required dotenv files exists.
```
Comment (ujvl) [inline sbin/start_sp_magi.sh:8]:
```suggestion
# Check that all required dotenv files exists.
```

File: sbin/create_genesis.sh [hunk 9]
```diff
@@ -1,19 +1,28 @@
...
 fi
-echo "Using dotenv: $ENV"
-. $ENV
-
-if [ ! -d "$OPS_DIR" ]; then
-    SBIN=`dirname $0`
-    SBIN="`cd "$SBIN"; pwd`"
-    . $SBIN/configure.sh
-    OPS_DIR="`cd "$OPS_DIR"; pwd`"
+echo "Using configure dotenv: $PATHS_ENV"
```
Comment (ujvl) [inline sbin/create_genesis.sh:15]:
```suggestion
echo "Using paths dotenv: $PATHS_ENV"
```

File: sbin/start_l1.sh [hunk 10]
```diff
@@ -1,50 +1,60 @@
...
 CONTRACTS_ENV=".contracts.env"
-if ! test -f $CONTRACTS_ENV; then
-    echo "Expected dotenv at $CONTRACTS_ENV (does not exist)."
+if ! test -f "$CONTRACTS_ENV"; then
+    echo "Expected contracts dotenv at $CONTRACTS_ENV (does not exist)."
     exit
 fi
 echo "Using contracts dotenv: $CONTRACTS_ENV"
 . $CONTRACTS_ENV
+fi
```
Comment (ujvl) [inline sbin/start_l1.sh:31]: why only for `geth`? also the indentation is off

File: sbin/export_genesis.sh [hunk 11]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
```
Comment (ujvl) [inline sbin/export_genesis.sh:1]: this file should be deleted?

File: sbin/create_genesis.sh [hunk 12]
```diff
@@ -1,19 +1,28 @@
...
+
+GENESIS_ENV=".genesis.env"
+if ! test -f "$GENESIS_ENV"; then
+    echo "Expected dotenv at $GENESIS_ENV (does not exist)."
+    exit
 fi
+echo "Using dotenv: $GENESIS_ENV"
+. $GENESIS_ENV
+
 echo "Using $OPS_DIR as ops directory."
```
Comment (ujvl) [inline sbin/create_genesis.sh:26]: this is deleted though..?

**Review by ujvl — CHANGES_REQUESTED** [review]

File: sbin/start_l1.sh (hunk 10 above)
Comment (siosw) [reply sbin/start_l1.sh:31]: only needed to get the addresses to fund, HH does this automatically
a lot of the indentation across all scripts is a bit off rn, #244 fixes this

File: sbin/create_genesis.sh (hunk 12 above)
Comment (siosw) [reply sbin/create_genesis.sh:26]: can you explain, will this be removed in other PR?

**Pull request merged**